webapp/channels/src/components/advanced_text_editor/use_submit.tsx

```diff
@@ -185,27 +190,27 @@ const useSubmit = (
             return;
         }
 
-        if (!postId) {
+        if (!postId && !schedulingInfo) {
             dispatch(scrollPostListToBottom());
         }
 
         isDraftSubmitting.current = false;
     }, [handleDraftChange, dispatch, draft, focusTextbox, isRootDeleted, postError, serverError, showPostDeletedModal, channelId, postId, lastBlurAt, setPostError, setServerError, afterSubmit]);
 
-    const showNotifyAllModal = useCallback((mentions: string[], channelTimezoneCount: number, memberNotifyCount: number) => {
+    const showNotifyAllModal = useCallback((mentions: string[], channelTimezoneCount: number, memberNotifyCount: number, schedulingInfo?: SchedulingInfo) => {
         dispatch(openModal({
             modalId: ModalIdentifiers.NOTIFY_CONFIRM_MODAL,
             dialogType: NotifyConfirmModal,
             dialogProps: {
                 mentions,
                 channelTimezoneCount,
                 memberNotifyCount,
-                onConfirm: () => doSubmit(),
+                onConfirm: () => doSubmit(undefined, undefined, schedulingInfo),
```

Comment:
Instead of having `showNotifyAllModal` depend on `doSubmit`, I think it might be better for it to take a callback that tells it what to do like `prioritySubmitCheck` does.

Related to my comment about having two actions for `submitPost` and `schedulePost` though, the fact that we're having to pass around and consider `schedulingInfo` here feels like a red flag, especially since we have to pass `undefined` through as the first two arguments

Reply:
Isn't this already taking a callback, via dialogProps? Callbacks to NotifyConfirmModal and prioritySubmitCheck are exactly the same. Or am I missing something.

Reply:
Sorry, I meant that we pass an `onConfirm` callback into `showNotifyAllModal` instead of having it call `doSubmit` directly like the one that `prioritySubmitCheck` takes. It still risks skipping followup logic like I mentioned below, but it saves us from having this callback depend on `doSubmit` and needing to take `schedulingInfo`. 0/5 though on actually changing it

Reply:
Ah I understand now. Done!